guide/app/layout.tsx

```diff
@@ -106,16 +172,30 @@ export function Guide({
 	return (
 		<div className="xl:container mx-auto xl:grid xl:grid-cols-5 gap-10 px-8 relative">
 			<header className="bg-zinc-900 xl:bg-transparent sticky top-0 max-h-screen z-10 flex flex-col">
-				<div className="py-2 xl:py-4">
+				<div className="py-2 xl:py-4 flex items-center">
 					<Link
-						className="font-mono inline-block py-4 text-[.25rem] leading-[.25rem] xl:text-[.35rem] xl:leading-[.40rem] whitespace-pre"
+						className="flex-1 font-mono inline-block py-4 text-[.25rem] leading-[.25rem] xl:text-[.35rem] xl:leading-[.40rem] whitespace-pre"
 						title="Conform"
 						to="/"
 					>
 						{logo}
 					</Link>
+					<select
+						className="bg-zinc-900 text-xs rounded-md py-1"
+						defaultValue={language.code}
+						onChange={(e) => {
+							const selectedLanguage = getLanguage(e.target.value);
+							window.location.href = `//${selectedLanguage.domain}${location.pathname}${location.search}`;
```

Comment:
should use `navigate` from useNavigate hook

Reply:
This place does not work unless window.location.href is used.

Transitions due to language switching are external URLs. The useNavigate/navigate of react-router supports external URLs and navigate does not allow page transitions.
The react-router author is told to use window.location.href in [this issue](
https://github.com/remix-run/react-router/issues/8317#issuecomment-974577787).

Reply:
@coji ah, sorry, I didn't look close enough, yes you are right!